channeldb/channel.go

```diff
@@ -2224,6 +2224,58 @@ type HTLC struct {
 	ExtraData []byte
 }
 
+// SetExtraData encodes a TLV stream of extra data to be stored with a HTLC.
+// It uses the update_add_htlc TLV types, because this is where extra data is
+// passed with a HTLC. At present blinding points are the only extra data that
+// we will store, and the function is a no-op if a nil blinding point is
+// provided.
+func (h *HTLC) SetExtraData(blindingPoint *btcec.PublicKey) error {
```

Comment:
> I assume the reason is to have the HTLC struct closely represent the schema used on disk.

Yeah, reasoning was to only have the fields that are persisted on the `channeldb.HTLC` struct, since we're already using a kinda odd workaround to store the data. 

> instead of just *BlindedPoint and then just handling the persistence details in the HTLC encoding/decoding methods

Looking at this again, this seems to be the approach that we take for messages with TLVs like `AcceptChannel`, so would be happy to update to match that? Could also pair it with un-exporting `ExtraData` on `channeldb.HTLC` which could be a little cleaner 🤔